### PR TITLE
[ENH] Added `node name` to the dataset result card

### LINKED ISSUE #279: On a dataset card, show which federation node it is coming from
On a dataset result card, show some indicator of what federation node it is coming from.
data_card

related: https://github.com/neurobagel/query-tool/issues/267

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #279 

Changes proposed in this pull request:
- Applied the `v-for` to `b-button` instead of `b-button-toolgroup` in `ResultCard`

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [x] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: components/ResultCard.vue [hunk 1]
```diff
@@ -53,18 +54,22 @@
           <b-col cols="4">
             <b-row>
               <b-button-toolbar
-                class="d-flex justify-content-end"
+                class="d-flex flex-column align-items-end"
               >
+                <span
+                  class="text-right mb-2"
+                  :data-cy="`card-${datasetUuid}-node-name`"
+                >{{ `from ${nodeName}` }}</span>
```
Comment (surchs) [inline components/ResultCard.vue:62]: nice!

File: components/ResultCard.vue [hunk 2]
```diff
@@ -53,18 +54,22 @@
...
                 <b-button-group
-                  v-for="modality in sortedModalities"
-                  :key="modality"
-                  class="mr-2"
+                  class="float-right"
                 >
                   <b-button
+                    v-for="modality in sortedModalities"
+                    :key="modality"
```
Comment (surchs) [inline components/ResultCard.vue:68]: also nice! Could you please add that to the PR description (I don't think we had an issue for that yet)

File: cypress/component/ResultCard.cy.js [hunk 3]
```diff
@@ -10,15 +11,16 @@ const props = {
...
       propsData: props,
     });
+    cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-node-name"]').should('be.visible').contains('some-node-name');
     cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-dataset"]').should('be.visible').contains('cool-dataset');
     cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-subjects"]').should('be.visible').contains('10 total subjects');
     cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-subjects"]').should('be.visible').contains('4 subjects');
-    cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-T1"]').should('be.visible').contains('T1');
-    cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-T2"]').should('be.visible').contains('T2');
+    cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-modality-T1"]').should('be.visible').contains('T1');
+    cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-modality-T2"]').should('be.visible').contains('T2');
```
Comment (surchs) [inline cypress/component/ResultCard.cy.js:23]: I think the test here would be to see if they are sorted, right? picking by modality name and then asserting on modality name seems a bit circular. I think for now a TODO comment reminding us to assert sorting could be enough.

File: cypress/component/ResultsContainer.cy.js [hunk 4]
```diff
@@ -115,12 +115,12 @@ describe('Results', () => {
     cy.get('[data-cy="example-usage-modal"]').should('contain', 'dataset-level-results.tsv');
     cy.get('[data-cy="example-usage-modal"]').should('contain', 'participant-level-results.tsv');
   });
-  it.only('Displays the modality buttons in correct order', () => {
```
Comment (surchs) [inline cypress/component/ResultsContainer.cy.js:118]: :eyes: users hate this one simple trick to get your tests to pass

File: cypress/component/ResultCard.cy.js (hunk 3 above)
Comment (surchs) [reply cypress/component/ResultCard.cy.js:23]: aha, upon reading further to the ResultsContainer test, I am now confused where the test for "being sorted" should live. But am happy with Container too!

**Review by surchs — APPROVED** [review]
very nice! :cook:

File: components/ResultCard.vue (hunk 2 above)
Comment (rmanaem) [reply components/ResultCard.vue:68]: Yea just noticed it when I was working on the layout

File: cypress/component/ResultCard.cy.js (hunk 3 above)
Comment (rmanaem) [reply cypress/component/ResultCard.cy.js:23]: Yea I reckon we discussed this in the PR where we ordered the modalities.

**Pull request merged**